src/main/java/teammates/storage/sqlentity/Instructor.java

```diff
@@ -0,0 +1,237 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import com.googlecode.objectify.annotation.Entity;
+import com.googlecode.objectify.annotation.Id;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.PrimaryKeyJoinColumn;
+import jakarta.persistence.Table;
+import teammates.ui.output.InstructorPermissionRole;
+
+/**
+ * Represents an Instructor entity.
+ */
+@Entity
+@Table(name = "Instructors")
+// Might have to change this. Should follow User's PK. Same for Student
+@PrimaryKeyJoinColumn(name = "userId")
+public class Instructor { // TODO: extends User
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private int id;
```

Comment:
ID should be in User table. I've edited the ER diagram a little, hopefully this helps.